### PR TITLE
Feat/albedo wip

### PR DESCRIPTION
**Pull request by pgierz**
@mandresm, here is a cleaned up version of the albedo YAML. Maybe we can also briefly discuss the status so far, and if I need to install anything else via Spack.

### CODE REVIEW

File: configs/machines/albedo.yaml [hunk 1]
```diff
@@ -1,138 +1,180 @@
...
 operating_system: "linux-rocky"
-# Should not be defined here:
-#jobtype: compute NOTE(PG): ...why?? The computer configuration should not need to know anything about the jobtype.
 sh_interpreter: "/usr/bin/bash"
-
-
 # Batch system and configuration:
 batch_system: "slurm"
-accounting: false
+accounting: true
```
Comment (mandresm) [inline configs/machines/albedo.yaml:13]: I did not know Albedo will have accounts for the computing time... good to know.

File: configs/machines/albedo.yaml [hunk 2]
```diff
@@ -1,138 +1,180 @@
...
+            gcc:
+              HDF5ROOT: /albedo/soft/sw/spack-sw/hdf5/1.12.2-rgostku
+            intel-oneapi:
+              HDF5ROOT: /albedo/soft/sw/spack-sw/hdf5/1.12.2-fknnmx5
+        "intel-oneapi-mpi":
+          choose_compiler_suite:
+            gcc: 
+              HDF5ROOT: /albedo/soft/sw/spack-sw/hdf5/1.12.2-lnfmhhc
+            intel-oneapi:
+              HDF5ROOT: /albedo/soft/sw/spack-sw/hdf5/1.12.2-mt53hvh
```
Comment (mandresm) [inline configs/machines/albedo.yaml]: Are these choose blocks resolved?

**Review by mandresm — CHANGES_REQUESTED** [review]
Didn't we agree on indentation to be 4 spaces? Other than that, good to go from my side :)

**Comment by pgierz** [comment]
> Didn't we agree on indentation to be 4 spaces? Other than that, good to go from my side :)

Yeah, I have not correctly configured my vim yet on albedo :-(

File: configs/machines/albedo.yaml (hunk 2 above)
Comment (pgierz) [reply configs/machines/albedo.yaml]: Well, I just made a compile script with that version, so it seems to work. I'd rather have those variables set when you load the modules.

File: configs/machines/albedo.yaml (hunk 1 above)
Comment (pgierz) [reply configs/machines/albedo.yaml:13]: Just for reporting purposes; basically we want to have some way to reliably figure out how much compute time is being used by which group.

**Comment by mandresm** [comment]
Are now gonna be using the gcc versions? I thought you said it was going to be the intel-oneapi compiler ones

**Comment by mandresm** [comment]
Just checked, the `choose_` blocks inside the `export_vars` are not resolved. I think ESM-Tools only resolves choose blocks when they are at the base of a section (1st level of indentation). We should probably resolve the HDFROOT stuff outside into a variable. Do you want me to take care of that?

File: configs/machines/albedo.yaml (hunk 1 above)
Comment (mandresm) [reply configs/machines/albedo.yaml:13]: so there you would just write your awi user right? not your project?

File: configs/machines/albedo.yaml (hunk 2 above)
Comment (mandresm) [reply configs/machines/albedo.yaml]: they don't work, see comment below

Comment (pgierz) [reply configs/machines/albedo.yaml]: Sorry....monday morning. I was using an old compile script 🤦🏻‍♂️

**Comment by pgierz** [comment]
> Just checked, the `choose_` blocks inside the `export_vars` are not resolved. I think ESM-Tools only resolves choose blocks when they are at the base of a section (1st level of indentation). We should probably resolve the HDFROOT stuff outside into a variable. Do you want me to take care of that?

yes please

File: configs/machines/albedo.yaml [hunk 3]
```diff
@@ -123,58 +124,17 @@ export_vars:
...
 
-  # NetCDF environment Variables:
+  # HDF5 exported environment variables:
+  HDF5ROOT: "$(module show hdf5/${hdf5_version}-${mpi_implementation_suffix}-${compiler_suffix} | CMAKE_PREFIX_PATH | cut -d " " -f 2)"
+  HDF5_C_INCLUDE_DIRECTORIES: $HDF5ROOT/include
+  HDF5_ROOT: $HDF5ROOT
+
+  # NetCDF environment variables:
+  NETCDFFROOT: "$(module show netcdf-fortran/${netcdf_fortran_version}-${mpi_implementation_suffix}-${compiler_suffix} | CMAKE_PREFIX_PATH | cut -d " " -f 2)"
+  NETCDFROOT: "$(module show netcdf-c/${netcdf_c_version}-${mpi_implementation_suffix}-${compiler_suffix} | CMAKE_PREFIX_PATH | cut -d " " -f 2)"
```
Comment (pgierz) [inline configs/machines/albedo.yaml]: Still needs a `grep`?

Comment (mandresm) [reply configs/machines/albedo.yaml]: yep, commitng now

**Comment by pgierz** [comment]
I also fixed the idiotic broken Python 3.6 test here. Seems we were using an ancient version of the setup Python action, plus Python 3.6 had end-of-support beginning of 2022.

**Comment by pgierz** [comment]
You're kidding me. I forgot a second F?!

**Comment by mandresm** [comment]
That doesn't fix our netcdf problems though :(

**Review by mandresm — APPROVED** [review]

**Comment by mandresm** [comment]
I'm merging this one now, so that we can resume work from the `feat/albedo` branch

**Pull request merged**